worker/logsender/worker.go

```diff
@@ -53,7 +53,6 @@ func New(logs LogRecordCh, logSenderAPI *logsender.API) worker.Worker {
 		case <-stop:
 			return nil
 		}
-		defer logWriter.Close()
```

Comment:
If we were closing it here, were we actually leaking anything?

Reply:
This one doesn't fix a possible leak but a possible panic:

The logWriter lifetime is already handled in another goroutine:

https://github.com/juju/juju/pull/18275/files/c9639e29c792e0b4d35fde55ece792c638e73919#diff-67753ed0473ba48fbfd61f6cc8431b39681c23a7bdb7e2b126ce1f83d3d10f1fR32-R46

The close function in the go routine has been added in a distinct commit, years later, to fix some strange behavior.

I assume that the author just miss to remove this Close.

As far I have understand the workload, this logWriter is retrieved from a local channel which is fed by the goroutine, which monitor the lifetime of the logWriter already.

Let me know if I miss something or if my explanation aren't clear enough.

Reply:
Oh actually I miss something.

I tried to limit the changes in the code, but the right way to do it is to remove this line and defer the close in the above goroutine.

Reply:
Yep, thanks for your careful check.

I have actually misunderstood something. I restored the code and add few comments (if I made I mistake, it is because it wasn't clear enough).

I thought about more elegant way to do it, but don't find an elegant way to do so without changing too much code. So the comments.